pkg/operator/metrics/status_metrics.go

```diff
@@ -0,0 +1,118 @@
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+func addStatusConditionsMetrics(c *Collector) {
+	c.register(
```

Comment:
none of these uses `c` other than for calling `register`. Why not make them independent funcs and call `c.register` at a central place. Would make the code easier and maybe help with solving this "prometheus panic on second call" problem.